searchlib/src/vespa/searchlib/queryeval/orlikesearch.h

```diff
@@ -66,5 +66,86 @@ class OrLikeSearch : public OrSearch
     Unpack _unpacker;
 };
 
+template <typename Unpack, typename HEAP, typename ref_t>
+class StrictHeapOrSearch : public OrSearch
+{
+private:
+    struct Less {
+        const uint32_t *child_docid;
+        explicit Less(const std::vector<uint32_t> &cd) : child_docid(cd.data()) {}
+        bool operator()(const ref_t &a, const ref_t &b) const {
+            return (child_docid[a] < child_docid[b]);
+        }
+    };
+
+    std::vector<uint32_t> _child_docid;
+    std::vector<ref_t>    _data_space;
+    ref_t                *_data_begin;
```

Comment:
I would move the most frequently used member to the front. Relative addressing has a higher cost for larger offsets. Used to be 64 or 128 on intel I think.